packages/Search/src/DTK_DistributedSearchTree_decl.hpp

```diff
@@ -115,8 +116,7 @@ void DistributedSearchTree<DeviceType>::query(
 {
     using Tag = typename Query::Tag;
     Details::DistributedSearchTreeImpl<DeviceType>::queryDispatch(
-        _comm, _top_tree, _bottom_tree, queries, indices, offset, ranks,
-        Tag{} );
+        *this, queries, indices, offset, ranks, Tag{} );
```

Comment:
If you need to make `DistributedSearchTreeImpl` a friend and pass `*this`, why not move `queryDispatch` to the class. Because this is basically what you are doing but in a complicated way.

Reply:
Another motivation for that was, if I make it a member of the class, I will not be able to mark it as private if I want to use Kokkos lambdas.